code/_hooks/events/entered.dm

```diff
@@ -0,0 +1,26 @@
+//  Ported from Baystation with permission from CrimsonShrike
+//  https://github.com/Baystation12/Baystation12/pull/30310
+//
+//	Observer Pattern Implementation: Entered
+//		Registration type: /atom
+//
+//		Raised when: An /atom/movable instance has entered an atom.
+//
+//		Arguments that the called proc should expect:
+//			/atom/entered: The atom that was entered
+//			/atom/movable/enterer: The instance that entered the atom
+//			/atom/old_loc: The atom the enterer came from
+//
+
+var/event/entered/entered_event = new()
```

Comment:
We don't need to create instances of events, pass the `/event/entered` path to the (un)register_event procs instead.